particles/tlv/particle.py

```diff
@@ -7,9 +7,12 @@
 import math
 
 class Particle(BaseParticle, RootObj):
-    def __init__(self, pdgid, charge, tlv, status=1):
+    def __init__(self, pdgid, charge, tlv, status=1, index=None, subtype='u'):
         super(Particle, self).__init__()
         self._pid = pdgid
+        self._uid = 0
+        if index != None:
+            self._uid = Identifier.make_id(Identifier.PFOBJECTTYPE.PARTICLE, index, subtype, tlv.E())
```

Comment:
_uid should be set properly in all cases, maybe get it in some form from RootObj?